network/upload.go

```diff
@@ -0,0 +1,48 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-utils/v2/retryhttp"
```

Comment:
I noticed you use the go-utils retryhttp module here, but use the `github.com/hashicorp/go-retryablehttp` one everywhere else. Any reason?

Reply:
I'm using the go-utils package to create a client instance and it returns a `retryablehttp.Client`. The reason I used other methods from the `retryablehttp` package (versus `net/http`) is that I noticed that it has some convenience methods compared to `net/http`. For example, `retryablehttp.NewRequest()` takes the request body as an `interface{}` and figures out what to do with it, while `net/http.NewRequest()` needs an `io.Reader` strictly.